wsmaster/che-core-api-system/src/main/java/org/eclipse/che/api/system/server/SystemEventsWebsocketBroadcaster.java

```diff
@@ -54,13 +67,41 @@ public void onEvent(SystemEvent event) {
     private static SystemEventDto asDto(SystemEvent event) {
         switch (event.getType()) {
             case STATUS_CHANGED:
-                SystemStatusChangedEvent statusChanged = (SystemStatusChangedEvent)event;
-                return DtoFactory.newDto(SystemStatusChangedEventDto.class)
-                                 .withStatus(statusChanged.getStatus())
-                                 .withPrevStatus(statusChanged.getPrevStatus())
-                                 .withType(statusChanged.getType());
+                return DtoFactory.newDto(SystemStatusChangedEventDto.class).copy((SystemStatusChangedEvent)event);
+            case SERVICE_ITEM_STOPPED:
+                return DtoFactory.newDto(SystemServiceItemStoppedEventDto.class).copy((SystemServiceItemStoppedEvent)event);
+            case SERVICE_STOPPED:
+            case STOPPING_SERVICE:
+                return DtoFactory.newDto(SystemServiceEventDto.class).copy((SystemServiceEvent)event);
             default:
-                throw new IllegalArgumentException("Can't convert event of type '" + event.getType() + "' to DTO");
+                throw new IllegalStateException("Can't convert event to dto, event type '" + event.getType() + "' is unknown");
+        }
+    }
+
+    /** An abstraction that allows to switch message sender backend. */
+    @VisibleForTesting
+    interface MessageSender {
+
+        /**
+         * Sends a given text message to a given channel.
+         *
+         * @param channel
+         *         channel to send message to
+         * @param message
+         *         message to send
+         * @throws Exception
+         *         when any error occurs
+         */
+        void sendMessage(String channel, String message) throws Exception;
+    }
+
+    private static class WebsocketMessageSender implements MessageSender {
```

Comment:
We already have pretty the same class org.eclipse.che.api.core.util.WebsocketMessageConsumer. Please consider usage of it.